forms-flow-web/src/services/FileService.js

```diff
@@ -33,27 +33,45 @@ const uploadFile = (evt, callback) => {
   }  
 };
 
-const extractFormDetails = (fileContent, callback) => {
+const extractFileDetails = (fileContent, callback) => { 
   const fileObj = fileContent; // The file object passed directly
   if (fileObj) {
     const reader = new FileReader(); // Initialize FileReader to read the file
     reader.readAsText(fileObj); // Read the file as text
 
     reader.onload = (e) => {
       try {
-        const fileContents = JSON.parse(e.target.result); // Parse file content as JSON
+        const fileExtension = fileObj.name.split('.').pop().toLowerCase();
+        
+        if (fileExtension === 'json') {
+          // Handle JSON file parsing
+          const fileContents = JSON.parse(e.target.result);
 
-        // Check if 'forms' exist and is an array in fileContents
-        if (fileContents && fileContents.forms && Array.isArray(fileContents.forms)) {
-          const formToUpload = fileContents.forms[0]; // Extract the first form (or adjust as needed)
-          callback(formToUpload); // Pass the extracted form details to the callback function
+          // Check if 'forms' exist and is an array in fileContents
+          if (fileContents && fileContents.forms && Array.isArray(fileContents.forms)) {
+            const formToUpload = fileContents.forms[0]; // Extract the first form
+            callback(formToUpload); // Pass the extracted form details to the callback function
+          } else {
+            console.error("No 'forms' array found in the file.");
+            callback(null); // Pass null if the 'forms' array is missing
+          }
+        } else if (['bpmn', 'dmn'].includes(fileExtension)) {
+          // Handle XML file parsing
+          const parser = new DOMParser();
+          const xmlDoc = parser.parseFromString(e.target.result, "application/xml");
+
+          if (xmlDoc.getElementsByTagName("parsererror").length > 0) {
```

Comment:
do a null check . if xmlDoc.getElementsByTagName("parsererror") is not present, then this line will break